### ISSUE TITLE
Cannot install Bolt 1.2.6

### ISSUE THREAD

**Issue by buzzard2** [BoltEngine/Bolt-Tracker#110]
### How to reproduce issue
1. Create empty Unity project
2. Import Bolt 1.2.6
3. In Wizard window click "Install core Bolt package"

### Expected Behavior
Installation finishes successfully 

### Actual behavior
I receive two errors:

`Couldn't create asset file!
UnityEditor.AssetDatabase:CreateAsset()
BoltWizardWindow:InitialSetup() (at Assets/bolt/scripts/editor/BoltWizardWindowUtils.cs:273)
BoltWizardWindow:Install(BoltPackage) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:924)
<>c__DisplayClass62_0:<DrawInstallOption>b__0() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:800)
BoltWizardWindow:DrawStepOption(Texture2D, GUIContent, GUIContent, Nullable`1, Action, Action) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:869)
BoltWizardWindow:DrawInstallOption(BoltInstalls) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:817)
BoltWizardWindow:DrawSetupBolt() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:485)
BoltWizardWindow:DrawContent() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:445)
BoltWizardWindow:OnGUI() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:421)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)`

`UnityException: Creating asset at path [PROJECT_PATH]\Assets\bolt\resources\BoltRuntimeSettings.asset failed.
BoltWizardWindow.InitialSetup () (at Assets/bolt/scripts/editor/BoltWizardWindowUtils.cs:273)
BoltWizardWindow.Install (BoltWizardWindow+BoltPackage package) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:924)
BoltWizardWindow+<>c__DisplayClass62_0.<DrawInstallOption>b__0 () (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:800)
BoltWizardWindow.DrawStepOption (UnityEngine.Texture2D icon, UnityEngine.GUIContent header, UnityEngine.GUIContent description, System.Nullable`1[T] active, System.Action callback, System.Action ignoredCallback) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:869)
BoltWizardWindow.DrawInstallOption (BoltWizardWindow+BoltInstalls install) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:817)
BoltWizardWindow.DrawSetupBolt () (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:485)
BoltWizardWindow.DrawContent () (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:445)
BoltWizardWindow.OnGUI () (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:421)
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <d7ac571ca2d04b2f981d0d886fa067cf>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <d7ac571ca2d04b2f981d0d886fa067cf>:0)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <d7ac571ca2d04b2f981d0d886fa067cf>:0)
UnityEditor.HostView.Invoke (System.String methodName, System.Object obj) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:342)
UnityEditor.HostView.Invoke (System.String methodName) (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:336)
UnityEditor.HostView.OldOnGUI () (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:129)
UnityEngine.Experimental.UIElements.IMGUIContainer.DoOnGUI (UnityEngine.Event evt, UnityEngine.Matrix4x4 worldTransform, UnityEngine.Rect clippingRect, System.Boolean isComputingLayout) (at C:/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:266)
UnityEngine.Experimental.UIElements.IMGUIContainer.HandleIMGUIEvent (UnityEngine.Event e, UnityEngine.Matrix4x4 worldTransform, UnityEngine.Rect clippingRect) (at C:/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:438)
UnityEngine.Experimental.UIElements.IMGUIContainer.HandleIMGUIEvent (UnityEngine.Event e) (at C:/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:421)
UnityEngine.Experimental.UIElements.IMGUIContainer.HandleEvent (UnityEngine.Experimental.UIElements.EventBase evt) (at C:/buildslave/unity/build/Modules/UIElements/IMGUIContainer.cs:401)
UnityEngine.Experimental.UIElements.EventDispatcher.PropagateEvent (UnityEngine.Experimental.UIElements.EventBase evt) (at C:/buildslave/unity/build/Modules/UIElements/EventDispatcher.cs:754)
UnityEngine.Experimental.UIElements.EventDispatcher.ProcessEvent (UnityEngine.Experimental.UIElements.EventBase evt, UnityEngine.Experimental.UIElements.IPanel panel) (at C:/buildslave/unity/build/Modules/UIElements/EventDispatcher.cs:599)
UnityEngine.Experimental.UIElements.EventDispatcher.Dispatch (UnityEngine.Experimental.UIElements.EventBase evt, UnityEngine.Experimental.UIElements.IPanel panel, UnityEngine.Experimental.UIElements.DispatchMode dispatchMode) (at C:/buildslave/unity/build/Modules/UIElements/EventDispatcher.cs:307)
UnityEngine.Experimental.UIElements.BaseVisualElementPanel.SendEvent (UnityEngine.Experimental.UIElements.EventBase e, UnityEngine.Experimental.UIElements.DispatchMode dispatchMode) (at C:/buildslave/unity/build/Modules/UIElements/Panel.cs:176)
UnityEngine.Experimental.UIElements.UIElementsUtility.DoDispatch (UnityEngine.Experimental.UIElements.BaseVisualElementPanel panel) (at C:/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:245)
UnityEngine.Experimental.UIElements.UIElementsUtility.ProcessEvent (System.Int32 instanceID, System.IntPtr nativeEventPtr) (at C:/buildslave/unity/build/Modules/UIElements/UIElementsUtility.cs:68)
UnityEngine.GUIUtility.ProcessEvent (System.Int32 instanceID, System.IntPtr nativeEventPtr) (at C:/buildslave/unity/build/Modules/IMGUI/GUIUtility.cs:179)
`

### Configuration
- Unity Version: 2018.3.6f
- Bolt Version: 1.2.6 free
- Operating System: win10

**Comment by ramonmelo** [comment]
Can you try to create your project in a different location? Maybe you don't have some permissions on the current directory you are trying to run the code.

The `BoltRuntimeSettings.asset` will only be recreated when running the `Wizard` Window if you delete it from the package (see line 268 of `BoltWizardWindowUtils.cs`), as the Bolt installation pack already comes with the default settings file.

Even to start the Wizard window, you need to have the `Assets/bolt/resources/BoltRuntimeSettings.asset` in place, so I don't see how this is possible to happen.

In any case, I've tried to reproduce your behavior several times using Unity 2017.x, 2018.1.x, 2018.2.x and 2018.3.x, but with no success.

Could you check if this occurs on other versions too?

Please, let me know.

**Comment by buzzard2** [comment]
This error occurs in 2018.2.20 too.
Changing directory and giving full permissions didn't help either

**Comment by DavidMann10k** [comment]
With the same configuration as @buzzard2 I'm seeing these errors and the same behavior described, but I thought I'd add that there's a  warning that I get at the same time: 

Invalid AssetDatabase path: C:\Workspace\Unity\TestBolt\Assets\bolt\resources\BoltRuntimeSettings.asset. Use path relative to the project folder.
UnityEditor.AssetDatabase:LoadAssetAtPath()
BoltWizardWindow:InitialSetup() (at Assets/bolt/scripts/editor/BoltWizardWindowUtils.cs:268)
BoltWizardWindow:Install(BoltPackage) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:924)
<>c__DisplayClass62_0:<DrawInstallOption>b__0() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:800)
BoltWizardWindow:DrawStepOption(Texture2D, GUIContent, GUIContent, Nullable`1, Action, Action) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:869)
BoltWizardWindow:DrawInstallOption(BoltInstalls) (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:817)
BoltWizardWindow:DrawSetupBolt() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:485)
BoltWizardWindow:DrawContent() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:445)
BoltWizardWindow:OnGUI() (at Assets/bolt/scripts/editor/BoltWizardWindow.cs:421)
UnityEngine.GUIUtility:ProcessEvent(Int32, IntPtr)

**Comment by ramonmelo** [comment]
We are working to solve this. Thx for the feedback.

**Comment by MBoffin** [comment]
I'm getting this same behavior as well, with the same warning and errors. I should mention that I'm getting two warnings, though. The second warning is identical to @DavidMann10k's, except it's for
```
UnityEditor.AssetDatabase:CreateAsset(Object, String)
BoltWizardWindow:InitialSetup() (at Assets/bolt/scripts/editor/BoltWizardWindowUtils.cs:273)
```
instead of just the one warning for `LoadAsset()`.

Using 2018.2.201f1.

**Comment by ramonmelo** [comment]
Fixed on Bolt v1.2.7.

Thanks for the feedback.
